packages/database/src/migrations/20210120051444-ResyncSubmissionsAfterRelease70-modifies-data.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+import { DatabaseChangeChannel } from '../DatabaseChangeChannel';
+import { markRecordsForResync } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const MIN_SURVEY_END_TIME = '2021-01-14';
+
+const selectSurveyResponsesForResync = async db => {
+  const { rows } = await db.runSql(`
+    SELECT survey_response.*
+    FROM survey_response
+    JOIN survey ON survey_response.survey_id = survey.id
+    LEFT JOIN dhis_sync_queue ON survey_response.id = record_id
+    WHERE
+      dhis_sync_queue.id IS NULL AND
+      survey_response.end_time > '${MIN_SURVEY_END_TIME}'`);
+  return rows;
+};
+
+const selectAnswersForResync = async db => {
+  const { rows } = await db.runSql(`
+    SELECT answer.*
+    FROM answer
+    JOIN survey_response ON answer.survey_response_id = survey_response.id
+    JOIN survey ON survey_response.survey_id = survey.id
+    LEFT JOIN dhis_sync_queue ON answer.id = record_id
+    WHERE
+      survey_response.end_time > '${MIN_SURVEY_END_TIME}' AND
+      survey.can_repeat = false AND -- repeating surveys are event based, don't need answers in sync queue
+      survey.code NOT IN ('CD3a', 'CD3b') AND -- non org unit entity surveys, event based
```

Comment:
There are more of these now I believe

Reply:
Although it's fairly safe as marking for resync still follows normal protocol so they will be filtered out before making it to the queue. Just a potential optimisation I believe.